crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -222,27 +212,21 @@ impl<A> PeerGossip<A>
 where
     A: AuthorityAPI + Send + Sync + 'static + Clone,
 {
-    pub fn new(
-        peer_name: AuthorityName,
-        active_authority: &ActiveAuthority<A>,
-        start_seq: Option<TxSequenceNumber>,
-    ) -> PeerGossip<A> {
+    pub fn new(peer_name: AuthorityName, active_authority: &ActiveAuthority<A>) -> PeerGossip<A> {
         // TODO: for validator gossip, we should always use None as the start_seq, but we should
         // consult the start_seq we retrieved from the db to make sure that the peer is giving
         // us new txes.
         let start_seq = match active_authority
             .follower_store
             .get_next_sequence(&peer_name)
         {
-            Err(e) => {
-                error!("Could not load next sequence from follower store, defaulting to None. Error: {}", e);
-                // It might seem like a good idea to return start_seq here, but if we are running
-                // as a full node start_seq will be Some(0), and if the gossip process is repeatedly
-                // restarting, we would in that case repeatedly re-request all txes from the
-                // beginning of the epoch which could DoS the validators we are following.
-                None
+            Err(_e) => {
+                // If there was no start sequence found for this peer, it is likely a new peer
```

Comment:
I stumbled on this code and I think the comment/logic are not correct, Err here does not mean sequence not found, it means storage error and should probably be just popped up the stack rather then handled as `0`.
Not found returned as `Ok(None)`. I am changing this code in my PR anyway so no action required, but let me know if I misunderstood something